bgpd/bgp_route.c

```diff
@@ -3160,7 +3158,7 @@ static bool bgp_lu_need_null_label(struct bgp *bgp,
 	    || new_select->sub_type == BGP_ROUTE_AGGREGATE
 	    || new_select->sub_type == BGP_ROUTE_REDISTRIBUTE)
 		goto need_null_label;
-	else if (new_select->extra &&
+	else if (new_select->extra && new_select->extra->num_labels &&
```

Comment:
I see this check pattern in many places, we might have a helper for this stuff?

Reply:
Added bgp_path_has_valid_label() for that